controllers/tournamentController.js

```diff
@@ -30,5 +31,58 @@ module.exports.updateTournamentById = async (req, res, next) => {
 module.exports.deleteTournamentById = async (req, res, next) => {
     let id = req.params.id;
     await Tournament.findByIdAndDelete(id);
+    return next();
+}
+
+module.exports.getAllRoundsByTournamentId = async (req, res, next) => {
+    let id = req.params.id;
+    let tournament = await Tournament.findById(id, {rounds:1});
+    res.locals.tournament = tournament;
+    return next();
+}
+
+module.exports.getOneRoundFromTournamentById = async (req, res, next) => {
+    let tournamentId = req.params.id;
+    let roundId = req.params.round;
+
+    let tournament = await Tournament.findById(tournamentId, {rounds:1});
+    res.locals.round = tournament.rounds.id(roundId);
+
+    return next();
+}
+
+module.exports.updateRoundById = async (req, res, next) => {
+    let tournamentId = req.params.id;
+    let roundId = req.params.round;
+    let body = req.body;
+
+    let tournament = await Tournament.findOneAndUpdate(
+        { _id: tournamentId, 'rounds._id': roundId }, //search keys
+        { $set: { 'rounds.$': body } }
+    );
+    res.locals.tournament = tournament;
+        
+    return next();
+}
+
+module.exports.deleteRoundById = async (req, res, next) => {
+    let tournamentId = req.params.id;
+    let roundId = req.params.round;
+
+    let tournament = await Tournament.findById(id);
+    tournament.rounds.remove(roundId);
+
```

Comment:
You forgot to add the save function
```suggestion
await tournament.save();
```